### PR TITLE
Allow using custom IO devices in logger_std_h

### PR DESCRIPTION
**Pull request by hauleth**
This is needed for simpler integration with Elixir logger which allows any device to be set as an output device. It also provides greater flexibility when needed to integrate with log catchers.

### CODE REVIEW

**Comment by garazdawi** [comment]
I think that the type you want to include is `io:device()`. The `file:io_device()` type includes what is returned by `file:open(Path,[raw])` which I don't think will work in this case.

**Comment by hauleth** [comment]
Yeah, that was probably the type I wanted. I will update the data.

File: lib/kernel/doc/src/logger_std_h.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
        is stored in a sub map with the key <c>config</c>, and can contain the
        following parameters:</p>
     <taglist>
-      <tag><marker id="type"/><c>type = standard_io | standard_error | file</c></tag>
+      <tag><marker id="type"/><c>type = standard_io | standard_error | file | {device, atom() | io:device()}</c></tag>
```
Comment (garazdawi) [inline lib/kernel/doc/src/logger_std_h.xml]: `io:device()` already includes `atom()`, so no need to include it in the list.

Comment (hauleth) [reply lib/kernel/doc/src/logger_std_h.xml]: Now it should be fixed.

**Pull request merged**